trading_calendars/calendar_helpers.py

```diff
@@ -25,14 +28,31 @@ def previous_divider_idx(dividers, minute_val):
     return divider_idx - 1
 
 
-def is_open(opens, closes, minute_val):
+def is_open(opens, break_starts, break_ends, closes, minute_val):
 
     open_idx = np.searchsorted(opens, minute_val)
     close_idx = np.searchsorted(closes, minute_val)
 
     if open_idx != close_idx:
-        # if the indices are not same, that means the market is open
-        return True
+        # we are not guarenteed to have a sorted list of breaks, since break
+        # may not exist for a given date. Thus we need special logic.
+
+        # if the indices are not same, that means we are within a session
+        if (break_starts == NP_NAT).all() and (break_ends == NP_NAT).all():
+            # this calendar has no breaks
+            return True
```

Comment:
All these extra checks make me want us to build out breaks on every day, some off which may be null. Not sure the perf.